vars/withPactTestOnlyPipeline.groovy

```diff
@@ -0,0 +1,145 @@
+import uk.gov.hmcts.contino.AppPipelineConfig
+import uk.gov.hmcts.contino.AppPipelineDsl
+import uk.gov.hmcts.contino.PipelineCallbacksConfig
+import uk.gov.hmcts.contino.PipelineCallbacksRunner
+import uk.gov.hmcts.pipeline.AKSSubscriptions
+import uk.gov.hmcts.pipeline.deprecation.WarningCollector
+import uk.gov.hmcts.pipeline.TeamConfig
+
+def call(type, String product, String component, Closure body) {
+
+  def branch = new ProjectBranch(env.BRANCH_NAME)
+
+  def deploymentNamespace = branch.deploymentNamespace()
+  def deploymentProduct = deploymentNamespace ? "$deploymentNamespace-$product" : product
+
+  def pipelineTypes = [
+    java  : new SpringBootPipelineType(this, deploymentProduct, component),
+    nodejs: new NodePipelineType(this, deploymentProduct, component),
+    angular: new AngularPipelineType(this, deploymentProduct, component)
+  ]
+
+  Subscription subscription = new Subscription(env)
+  AKSSubscriptions aksSubscriptions = new AKSSubscriptions(this)
+
+  PipelineType pipelineType
+
+  if (type instanceof PipelineType) {
+    pipelineType = type
+  } else {
+    pipelineType = pipelineTypes.get(type)
+  }
+
+  assert pipelineType != null
+
+  MetricsPublisher metricsPublisher = new MetricsPublisher(this, currentBuild, product, component, subscription.prodName )
+  def pipelineConfig = new AppPipelineConfig()
+  def callbacks = new PipelineCallbacksConfig()
+  def callbacksRunner = new PipelineCallbacksRunner(callbacks)
+
+  callbacks.registerAfterAll { stage ->
+    metricsPublisher.publish(stage)
+  }
+
+  def dsl = new AppPipelineDsl(this, callbacks, pipelineConfig)
+  body.delegate = dsl
+  body.call() // register pipeline config
+
+  dsl.onStageFailure() {
+    currentBuild.result = "FAILURE"
+  }
+
+  Environment environment = new Environment(env)
+
+  node {
+    def slackChannel = new TeamConfig(this).getBuildNoticesSlackChannel(product)
+    try {
+      env.PATH = "$env.PATH:/usr/local/bin"
+
+      onMaster {
+
+        PipelineCallbacksRunner pcr = callbacksRunner
+        AppPipelineConfig config = pipelineConfig
+        Builder builder = pipelineType.builder
+
+        def azSubscription = subscription.nonProdName
+        def pactBrokerUrl = environment.pactBrokerUrl
+        def acr
+        def dockerImage
+        def projectBranch
+        boolean noSkipImgBuild = true
+
+        stage('Checkout') {
+          pcr.callAround('checkout') {
+            checkoutScm()
+          }
+        }
+
+        stage("Build") {
+
+          builder.setupToolVersion()
+        }
+
+        stage("Tests") {
+
+          when(noSkipImgBuild) {
+                pcr.callAround('test') {
+                  timeoutWithMsg(time: 20, unit: 'MINUTES', action: 'test') {
+                    builder.test()
+                  }
+                }
+              failFast: true
+          }
+        }
+
+        stage("Pact Consumer Verification") {
+          if (config.pactBrokerEnabled) {
```

Comment:
Is this check really needed? If someone is using this pipeline , don't we expect them to have pact tests anyway.